src/sa_web/static/js/views/app-view.js

```diff
@@ -306,6 +318,25 @@ var Shareabouts = Shareabouts || {};
           attribute: 'properties',
 
           success: function() {
+            // generate an attribute with relevant story information
+            // NOTE: right now a single model can only be part of a single story.
+            collection.forEach(function(model, index) {
+              model.set("story", function() {
+                var storyObj = null,
+                url = model.attributes.datasetSlug + "/" + model.attributes.id;
+                _.each(self.options.storyConfig, function(story) {
+                  if (story.order[url]) {
+                    storyObj = {
+                      tagline: story.tagline,
+                      next: story.order[url].next,
+                      previous: story.order[url].previous
+                    }
+                  }
+                });
+                return storyObj;
+              }());
+            });
```

Comment:
It seems inefficient to iterate through all of our models right after we fetch them. I wonder if we can refactor this functionality by setting our story attributes inside our Backbone PlaceModel's [parse method](http://backbonejs.org/#Model-parse)?

Here is our PlaceModel's parse method: https://github.com/smartercleanup/platform/blob/develop/src/sa_web/static/js/models.js#L27-L35 which sets everything in our `attributesToAdd` option that is passed into our BackboneCollection's `fetch` method. It seems like we can use this method to set our story attributes as well. It might require a little more logic to include our story config into the model attributes, and we may have to create another option like `storyAttributeToAdd`, but I think it would be cleaner and more efficient. But let me know what you think!

Also, our implementation is limited to places only, and it would be nice to have this story feature for our landmarks as well. Here is the parse method of our Backbone LandmarkModel: https://github.com/smartercleanup/platform/blob/develop/src/sa_web/static/js/models.js#L299-L309 If we can implement `parse` to set our PlaceModel's `story` attribute, we can do the same thing for our LandmarkModel's `story` attribute. Also, note the TODO comment in the LandmarkModel's Parse method linked above. If we can refactor the PlaceModel and LandmarkModel parse methods into the same parse function, then we would have a nice DRY implementation of setting our story config into our model attributes. Of course, although a perfectly DRY implementation would be nice, we can always refactor that on another implementation.

It would be great if someone can explore this possibility! I've only explored it a little bit, but I think it's possible.